app/src/main/java/com/example/ohthmhyh/activities/UpdateHabitEventActivity.java

```diff
@@ -102,6 +114,38 @@ public boolean onOptionsItemSelected(@NonNull MenuItem item) {
         }
         return super.onOptionsItemSelected(item);
     }
+    /**
+     * This handles the drop down list for Habits
+     * Can tell if we are on edit or new Calls to the data base to get the habitevent list
+     */
+    private void makeNameList(){
+        ArrayList<String> NameList=new ArrayList<>();
+        NameList.addAll(habitList.getHabitNames());
+        habitListAutoCompleteTextView = findViewById(R.id.AutoCompleteTextviewCE);
+        ArrayAdapter arrayAdapter = new ArrayAdapter(
+                this, R.layout.create_habit_habit_drop_down_menu, NameList);
+
+        Intent intent = getIntent();
+        habitEventIndex = intent.getIntExtra(ARG_HABIT_EVENT_INDEX, -1);
+        if (habitEventIndex >= 0) {
+            //Wait for habitEventList to become non null
+            databaseAdapter = new DatabaseAdapter();
+            databaseAdapter.pullHabitEvents(new DatabaseAdapter.HabitEventCallback() {
+                @Override
+                public void onHabitEventCallback(HabitEventList habitEvents) {
+                    habitEventList = habitEvents;
+                    for (int index=0;index<habitList.size();index++) {
+                        if (habitList.getHabit(index).getUHID() ==habitEventList.getHabitEvent(habitEventIndex).getHabitUHID()){
+                            habitListAutoCompleteTextView.setText(habitList.getHabit(index).getName(),false);
+                        }
+                    }
+                }
+            });
+
+        }else{habitListAutoCompleteTextView.setText(arrayAdapter.getItem(0).toString(),false);}
```

Comment:
This is hard to read. I suggest this:
```suggestion
        } else {
            habitListAutoCompleteTextView.setText(arrayAdapter.getItem(0).toString(), false);
        }
```

Reply:
Done